compiler/docgen.nim

```diff
@@ -222,6 +222,10 @@ proc getPlainDocstring(n: PNode): string =
       result = getPlainDocstring(n.sons[i])
       if result.len > 0: return
 
+func htmlCollapse(html: string): string =
+  ## Strip whitespace and remove trailing newlines from every line
+  return html.splitLines().mapIt(strip(it)).join("")
```

Comment:
Why is this required anyway?

Reply:
This implementation has terrible performance. ;-)

Reply:
I wrote it so I could format the HTML later in the code. Since that HTML ends up inside of a `<pre>`, it has to be collapsed.

The poor performance is from the string concatenation, yes? I should be using `result.add`?

Reply:
Actually, a [quick benchmark](https://gist.github.com/Quelklef/1f5a7a4c5208ac6ba2569aac4e9ffde2) on my machine shows a ~1.6x speedup using `result.add`, except when in `-d:release`, in which case it's about the same or slower! So I'm not totally sure what optimization you're looking for :P